### PR TITLE
Feat/i18next language provider

### PR DESCRIPTION
**Pull request by shamoilattaar-wednesday**
### Ticket Link
---------------------------------------------------

### Related Links
---------------------------------------------------

### Description
---------------------------------------------------

### Steps to Reproduce / Test
---------------------------------------------------

### GIF's
---------------------------------------------------

## Summary by CodeRabbit

- **New Features**
	- Transitioned the app's internationalization from Redux-based management to a simplified context-based approach using `react-i18next`.
	- Enhanced i18n setup with asynchronous language detection and improved translation handling.
	- Introduced a new utility function for conditional value selection to improve code readability.

- **Bug Fixes**
	- Updated tests to ensure compatibility with the new i18next library, enhancing reliability.

- **Documentation**
	- Updated comments and structures in various files to reflect changes in internationalization strategy.

- **Chores**
	- Added new dependencies for i18next and related libraries to improve internationalization support.

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
+import { I18nextProvider } from 'react-i18next';
 import LanguageProvider from '@atoms/LanguageProvider';
 import RootScreen from '@scenes/RootScreen';
+import i18n from '@app/i18n';
 import createStore from '@app/rootReducer';
-import { translationMessages } from './i18n';
-import 'react-native-gesture-handler';
 
+import { Provider } from 'react-redux';
+import { PersistGate } from 'redux-persist/lib/integration/react';
```
Comment (Anurag-Wednesday) [inline app/app.js:9]: can we please fix the import order

File: app/components/atoms/T/index.js [hunk 2]
```diff
@@ -6,15 +6,18 @@
...
-    {id ? intl.formatMessage({ id }, { ...values }) : text}
-  </Text>
-);
+import { useTranslation } from 'react-i18next';
+
+const T = ({ intl, id, values, style, text, ...otherProps }) => {
+  const { t } = useTranslation();
+  return (
+    <Text testID="t" style={style} {...otherProps}>
+      {id ? t(id, { ...values }) : text}
```
Comment (Anurag-Wednesday) [inline app/components/atoms/T/index.js:17]: Can we please add a conditional operator function here

File: app/app.js (hunk 1 above)
Comment (shamoilattaar-wednesday) [reply app/app.js:9]: placed as a placeholder for the other components that still are dependent on redux will be removed altogether

File: app/components/atoms/T/index.js (hunk 2 above)
Comment (shamoilattaar-wednesday) [reply app/components/atoms/T/index.js:17]: done

**Pull request merged**